src/uvwasi.c

```diff
@@ -127,6 +127,9 @@ void* uvwasi__malloc(const uvwasi_t* uvwasi, size_t size) {
 }
 
 void uvwasi__free(const uvwasi_t* uvwasi, void* ptr) {
+  if (uvwasi == NULL || uvwasi->allocator == NULL)
```

Comment:
I don’t think it’s okay to skip deallocating memory because we can’t look up the allocator – if we do have a non-null `ptr` to free, that means that it was produced by an allocator, therefore `uvwasi->allocator` should be accessible and non-null. This seems like it would make it *very* easy to introduce memory leaks.

If you want to do 

```suggestion
  if (ptr == NULL)
```

I think that should be fine.

Reply:
Thank you for suggestion, I change it.